### ISSUE TITLE
[Optimisation] Sort of Source Select without rescraping

### ISSUE THREAD

**Issue by DaisyF8** [Wilson-Magic/plugin.video.hummingbird#4]
If you search once for an episode (or several times to get more sources) you get a list in which you can manually select the source you want to try.  If you get an error and click play again, you also scrape again. With sometimes other sources than the previous srape, but it also takes time.

Can't you keep the previously found sources at hand, and allow a "Select from last search" ?

**Comment by Wilson-Magic** [comment]
I could make a caching of results for a while. Maybe an hour or so. But I don't want to keep them there for a long time. Anime sites are volatile with sources changing often. So you can never be sure.

And tbh I'm not entirely sure what you mean by that beta/alpha bit. But Thanks? I guess.

**Comment by DaisyF8** [comment]
This alpha release (you called it that on Reddit I think) is so good in my opinion, I'm wondering how you could top it in beta :) Nice work.

In case of watching a movie, if it stalls and you want to restart, maybe you're watching longer than an hour... So maybe keep it two hours ? But anyway, one hour is plenty for just finding a good source.

**Comment by Wilson-Magic** [comment]
Good point. Maybe 2 or 3 hours makes a bit more sense. I'll look into adding it to an update soon.

Thanks for the compliment though haha. I'd say it's more like a beta than an alpha. You don't want to know how bad this used to be. It used to take like 30 seconds to load one piece's episodes. And before that, it used to display lists of 20 episodes at a time, with no way to play all of them. So it was impossible to binge-watch. I also never used to cache any calls every single menu would take FOREVER lol.
